app/bank.py

```diff
@@ -126,173 +124,150 @@ def _get_bank_info(ota_config_file):
-                else:
-                    # device file name
-                    self._current_bank = fstab_list[0]
-                    if self._current_bank == self._banka:
-                        self._current_bank_uuid_str = "UUID=" + self._banka_uuid
-                        self._next_bank = self._bankb
-                        self._next_bank_uuid_str = "UUID" + self._bankb_uuid
-                    elif self._current_bank == self._bankb:
-                        self._current_bank_uuid_str = "UUID=" + self._bankb_uuid
-                        self._next_bank = self._banka
-                        self._next_bank_uuid_str = "UUID=" + self._banka_uuid
-                    else:
-                        self._read_fail = True
-                        logger.error("current bank is not banka or bankb!")
-                return fstab_list[0]
-
-        logger.info("root not found!")
-        return ""
+        self.bank_a, self.bank_b = _get_bank_info(self._bank_info_file)
+        self.bank_a_uuid = _get_uuid_from_blkid(self.bank_a)
+        self.bank_b_uuid = _get_uuid_from_blkid(self.bank_b)
 
     def get_banka(self):
         """
         Get bank A
         """
-        return self._banka
-
-    def get_banka_uuid(self):
-        """
-        Get bank A UUID
-        """
-        return self._banka_uuid
+        return self.bank_a
 
     def get_banka_uuid(self):
         """
         Get bank A UUID
         """
-        return self._banka_uuid
+        return self.bank_a_uuid
 
     def is_banka(self, bank):
         """
         Is bank A
         """
-        if bank == self._banka:
-            return True
-        return False
+        return bank == self.bank_a
 
     def is_banka_uuid(self, bank_uuid):
         """
         Is bank A UUID
         """
-        if bank_uuid == self._banka_uuid:
-            return True
-        return False
+        return bank_uuid == self.bank_a_uuid
 
     def get_bankb(self):
         """
         Get bank B
         """
-        return self._bankb
+        return self.bank_b
 
     def get_bankb_uuid(self):
         """
         Get bank B UUID
         """
-        return self._bankb_uuid
-
-    def get_bankb_uuid(self):
-        """
-        Get bank B UUID
-        """
-        return self._bankb_uuid
+        return self.bank_b_uuid
 
     def is_bankb(self, bank):
         """
         Is bank B
         """
-        if bank == self._bankb:
-            return True
-        return False
+        return bank == self.bank_b
 
     def is_bankb_uuid(self, bank_uuid):
         """
         Is bank B UUID
         """
-        if bank_uuid == self._bankb_uuid:
-            return True
-        return False
+        return bank_uuid == self.bank_b_uuid
+
+
+class BankInfo(_BaseBankInfo):
+    """
+    OTA Bank device info class
+    """
+
+    def __init__(self, *, fstab_file: str = None, bank_info_file: str = None):
+        # init current bank status
+        super().__init__()
```

Comment:
Actually what I think is that, default we use configuration values to configure which fstab_file and bank_info_file we should use, these configuration should not be changed in the production. So I initialize these values in the base class.
Values initialized in _BaseBankInfo should not be changed, and if we want to, we should do it in the child class BankInfo.

Reply:
b4d6bd1: fix call parent's init method before configuration change.

What I think is that, base class configures the BankInfo via configs.py. These configurations are not mean to be changed during actual run.
But if we want to manually indicate which fstab_file and bank_info_file to run in special cases(like run bank.py standalone), we should do it in the child class.

Reply:
```
bankinfo = BankInfo("my_fstab", "my_bank_info")
print(bankinfo.fstab_file, BankInfo.fstab_file)
print(bankinfo.bank_info_file, BankInfo.bank_info_file)
```
`bankinfo.fstab_file` and `BankInfo.fstab_file` are different. Is this intended? This may be difficult to understand.

Reply:
> ```
> bankinfo = BankInfo("my_fstab", "my_bank_info")
> print(bankinfo.fstab_file, BankInfo.fstab_file)
> print(bankinfo.bank_info_file, BankInfo.bank_info_file)
> ```
> 
> `bankinfo.fstab_file` and `BankInfo.fstab_file` are different. Is this intended? This may be difficult to understand.

Sorry but where these codes located? It is belonged to line 327~353 in bank.py?

Reply:
There is no code to access BankInfo.fstab_file. But base holds fstab_file as **class variable** and subclass also holds it as **instance variable**. This is not very clear.
I think we can do the following in the base class.
```python
def __init__(self, fstab_file=None, bank_info_file=None):
    if fstab_file:
        self.fstab_file = fstab_file
    else:
        self.fstab_file = cfg.BANK_INFO_FILE
    ...
```

Reply:
> There is no code to access BankInfo.fstab_file. But base holds fstab_file as **class variable** and subclass also holds it as **instance variable**. This is not very clear.
> I think we can do the following in the base class.
> 
> ```python
> def __init__(self, fstab_file=None, bank_info_file=None):
>     if fstab_file:
>         self.fstab_file = fstab_file
>     else:
>         self.fstab_file = cfg.BANK_INFO_FILE
>     ...
> ```

The idea is that, **fstab_file** and **bank_info_file** as **BaseBankInfo** class's variables are the default configuration(and the actual run configuration) for all the instances of **BankInfo**. 
In normal case, all the instances of **BankInfo** should not change these variables.
But in special case that we need to use custom **fstab_file** and **bank_info_file**, we should do that via that specific instance of BankInfo.

Reply:
OK I will come up with a new way to distribute the configurations to each classes, but I would like to implement it with a new PR.

Reply:
A new PR is OK!